analyses/focal-cn-file-preparation/05-define-most-focal-cn-units.Rmd

```diff
@@ -0,0 +1,415 @@
+if (!(running_in_ci)) {
+  # These are the cytobands that have not been defined as gain or loss --
+  # we want to define their gene level status
+  neutral_cytobands <- cytoband_status_count %>%
+    filter(dominant_status %in% c("unstable", "neutral")) %>%
+    dplyr::pull("cytoband")
+
+  # Filter the annotated CN data to include only these cytobands
+  neutral_status_cytoband_df <- combined_status_df %>%
+    filter(cytoband %in% neutral_cytobands)
+  
+  # Now count the gene level calls (for each status call) and define
+  # the gene as that status if more than 50% of the total counts are
+  # for that particular status
+  gene_status_count <- status_majority_caller(neutral_status_cytoband_df,
+                                              "gene_symbol",
+                                              "status", 
+                                              percent_threshold = percent_threshold)
+  
+  # Display table
+  gene_status_count
+}
+```
+
+## Plot calls
+
+Plot the final dominant status call on the x-axis and the percent of each status on the y-axis.
+
+### Plot chromosome arm status calls
+
+```{r}
+# Run `plot_dominant_status_calls` function for the chromosome arm calls
+plot_dominant_status_calls(
+  arm_status_count,
+  "chromosome_arm"
+)
+```
+
+### Plot cytoband status calls
+
+```{r fig.width = 10}
+# Run `plot_dominant_status_calls` function for the cytoband calls if not
+# running in CI
+if (!(running_in_ci)) {
+  plot_dominant_status_calls(cytoband_status_count,
+                             "cytoband")
+}
+```
+
+### Plot gene status calls
+
+```{r}
+# Plot if not running in circleCI
+if (!(running_in_ci)) {
+  plot_dominant_status_calls(gene_status_count,
+                             "gene_symbol")
+}
+```
+
+## Combine arm, cytoband, and gene level status data
+
+```{r}
+# The logic variable `running_in_ci` is needed here because the CI testing
+# files do not contain any of the genes in the `gene_status_count` data.frame
+# generated above (when `running_in_ci` == FALSE)
+if (!(running_in_ci)) {
+  final_df <- consensus_seg_autosomes_df %>%
+    mutate(chromosome_arm = gsub("(p|q).*", "\\1", cytoband)) %>%
+    inner_join(arm_status_count,
+               by = "chromosome_arm") %>%
+    left_join(cytoband_status_count,
+              by = "cytoband",
+              suffix = c(".arm", ".cytoband")) %>%
+    left_join(gene_status_count,
+              by = "gene_symbol",
+              suffix = c(".arm", ".gene")) %>%
+    mutate(
+      focal_call = paste0(
+        dominant_status.arm,
+        ", ",
+        dominant_status.cytoband,
+        ", ",
+        dominant_status
+      ),
+      # Here we want to define the most focal call based on the arm, cytoband,
+      # and gene status information -- if a loss/gain is defined at the arm
+      # level then the focal call will be "arm_loss" or "arm_gain" respectively,
+      # and so on.
+      focal_call = case_when(focal_call == "loss, NA, NA" ~ "arm_loss",
```

Comment:
I think we need to rework this last section here. In part because of something doesn't fall underneath the categories you lay out here then it automatically gets labeled as "Other" which is not a super helpful label.